src/lib/tls/msg_server_hello.cpp

```diff
@@ -30,7 +31,7 @@ const uint64_t DOWNGRADE_TLS11 = 0x444F574E47524400;
 const uint64_t DOWNGRADE_TLS12 = 0x444F574E47524401;
 
 // SHA-256("HelloRetryRequest")
-const std::array<uint8_t, 32> HELLO_RETRY_REQUEST_MARKER =
+const std::vector<uint8_t> HELLO_RETRY_REQUEST_MARKER =
```

Comment:
Why this change? Fine with it just curious why it was required.

Reply:
The c'tor of `Server_Hello_Internal` takes the random as a `std::vector` (which is usually provided from `make_hello_random()`, that returns a `std::vector`). Leaving this as a `std::array` would require a conversion or a c'tor overload just for that.

```C++

std::make_unique<Server_Hello_Internal>(.... , HELLO_RETRY_REQUEST_MARKER, ....);
// vs.
std::make_unique<Server_Hello_Internal>(.... , make_hello_random(), ....);
```

Or use a `std::span` in the `Server_Hello_Internal` c'tor? But wouldn't that cause UB because the `std::span` would refer to data in the temporary `std::vector` falling out of `make_hello_random()`?

My understanding is, that the temporary gets destroyed at the end of the expression (i.e. after the invoked c'tor returned). So as long as the c'tor copies the data out of the span, there should be no lifetime issues. [This quick test](https://godbolt.org/z/jTxKoW6cb) does support my view. Side note: An interesting article about views, ranges, life times and dangling pointers: https://tristanbrindle.com/posts/rvalue-ranges-and-views

But it would result in an unnecessary copy (from the temporary, via the span, into a member of `Server_Hello_Internal`). I think, my vote is to keep the `std::vector<>` for `HELLO_RETRY_REQUEST_MARKER`.